include/SFML/Window/Window.hpp

```diff
@@ -59,6 +59,29 @@ class SFML_WINDOW_API Window : GlResource, NonCopyable
 public:
 
     ////////////////////////////////////////////////////////////
+    /// \brief Enumeration of the native system cursor types
+    ///
+    ////////////////////////////////////////////////////////////
+    enum Cursor
+    {
+        Arrow,                  ///< Arrow cursor (default)
+        ArrowWait,              ///< Busy arrow cursor
+        Wait,                   ///< Busy cursor
+        Text,                   ///< I-beam, cursor when hovering over a field allowing text entry
```

Comment:
I personally think "Text" is fine as a name. It describes the cursor's purpose pretty well.